app/test/unit/git/for-each-ref-test.ts

```diff
@@ -58,4 +61,36 @@ describe('git/for-each-ref', () => {
       expect(status).toHaveLength(0)
     })
   })
+
+  describe('getBranchesDifferingFromUpstream', () => {
+    beforeEach(async () => {
+      const testRepoPath = await setupFixtureRepository(
+        'repo-with-non-updated-branches'
+      )
+      repository = new Repository(testRepoPath, -1, null, false)
+    })
+
+    it('filters branches differing from upstream using for-each-ref', async () => {
+      const allBranches = await getBranches(repository)
+
+      const branches = await getBranchesDifferingFromUpstream(
+        repository,
+        allBranches
+      )
+
+      const branchNames = branches.map(branch => branch.name)
+      expect(branchNames).toHaveLength(3)
+
+      // All branches that are behind and/or ahead must be included
+      expect(branchNames).toContain('branch-behind')
+      expect(branchNames).toContain('branch-ahead')
+      expect(branchNames).toContain('branch-ahead-and-behind')
+
+      // `master` is the current branch, and shouldn't be included
+      expect(branchNames).not.toContain('master')
```

Comment:
Is the local `master` branch behind its upstream? I.e. are we sure we're testing that it gets filtered out because it's the current branch and not just because it's up to date?

Also, I know this is a bit of a pain but do you think you could rename the branch in this fixture to `main` instead? We're trying to phase out the `master` references and have gotten rid of almost all of them save for tests.

Reply:
> Is the local `master` branch behind its upstream?
Yes!

I updated the repo with the branch rename and also added some info to its `README.md` to clarify that the `main` branch is the current branch and is behind its upstream.